### PR TITLE
I18n: Fix and reenable verify-i18n CI check

### PR DESCRIPTION
**Pull request by joshhunt**
This PR fixes outstanding issues that's causing the verify-i18n CI check to fail, and reenables it to fail builds:

 - Changes it to run on the `node:20-bookwork` image, which has `make` built into it
 - Fixes incorrect nav item string change in https://github.com/grafana/grafana/pull/84501
 - Extracts and commits current phrases so the verify step passes

### CODE REVIEW

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -644,13 +644,12 @@ def verify_i18n_step():
     uncommited_error_message = "\nTranslation extraction has not been committed. Please run 'make i18n-extract', commit the changes and push again."
     return {
         "name": "verify-i18n",
-        "image": images["node"],
+        "image": images["node_deb"],
```
Comment (joshhunt) [inline scripts/drone/steps/lib.star:647]: `node_deb` has `make` in it, so changed to that image.

File: scripts/drone/steps/lib.star [hunk 2]
```diff
@@ -644,13 +644,12 @@ def verify_i18n_step():
...
         "depends_on": [
             "yarn-install",
         ],
         "failure": "ignore",
         "commands": [
-            "apk add --update git",
```
Comment (joshhunt) [inline scripts/drone/steps/lib.star:653]: And `node_deb` already has `git` in it, so we can get rid of this also

**Comment by joshhunt** [comment]
PR build failing successfully because strings have not been extracted https://drone.grafana.net/grafana/grafana/176844/4/6

File: Makefile [hunk 3]
```diff
@@ -118,7 +118,7 @@ ifeq ("$(wildcard $(ENTERPRISE_FE_EXT_FILE))","") ## if enterprise is not enable
 i18n-extract-enterprise:
 	@echo "Skipping i18n extract for Enterprise: not enabled"
 else
-i18n-extract-enterprise: $(SWAGGER) ## Generate API Swagger specification
+i18n-extract-enterprise: ## Generate API Swagger specification
```
Comment (tskarhed) [inline Makefile:121]: What does this have to do with Swagger?

Comment (joshhunt) [reply Makefile:121]: oh, nothing. Just more copy-pasting i need to clean up 😶‍🌫️

**Review by tskarhed — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍 

~think we probably want _someone_ with CI knowledge (@grafana/grafana-backend-services-squad maybe?) to validate whether changing the image is ok~ nm, this is only for the `verify-i18n` step so much safer
